### PR TITLE
Fixed turret crash and ship overlap bug

### PR DESCRIPTION
**Pull request by Torpedo99**
# Description
Enables non engine ships to be spawned (like turrets). Also adds a fix for overlapping of ships when they spawn.

# Changes
- Unified `changeShip()` from `ChangeShipScreen.java` and `ChangeShipCommandHandler.java` as they were similar.
- Added a null check for engines in `changeShip()`
- Added an overlap check that makes sure the command is not run if the spawning ship overlaps another object.

### CODE REVIEW

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Uh oh, something went wrong with the build. Need to check on that

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: engine/src/main/java/org/destinationsol/game/Hero.java [hunk 1]
```diff
@@ -89,6 +97,58 @@ public void setSolShip(SolShip hero, SolGame solGame) {
...
 
+    public void changeShip(Hero hero, String newShipID, SolGame game, Console console) {
+        SolShip originalShip = hero.getShip();
+        Optional<SolShip> newShip;
+
+        boolean isARealShip = false;
+
+        for (ResourceUrn urn : Assets.getAssetHelper().list(Json.class)) {
+            if ((urn.getModuleName() + ":" + urn.getResourceName()).equals(newShipID)) {
+                isARealShip = true;
```
Comment (NicholasBatesNZ) [inline engine/src/main/java/org/destinationsol/game/Hero.java]: This here is wrong. What it's really doing is checking whether the given string exists as a Gestalt asset. When it's anything but a ship it causes issues down the line. I know it wasn't your code, just pointing out the issue. :+1:

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by NicholasBatesNZ** [comment]
Closing due to #473 :+1:

**Pull request closed without merging**